### PR TITLE
8 pack

### PR DESCRIPTION
**Pull request by gorbachev**
## Description

Adding an open source 2x4 8-Pack Macropad from https://github.com/cgarcia2097/8-Pack

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/8pack/rules.mk [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = no  # Virtual DIP switch configuration(+1000)
```
Comment (e11i0t23) [inline keyboards/8pack/rules.mk]: If there isn't a hardware reset switch, or it's not easily accessible, this should be changed to use Bootmagic Lite.

If not, this is fine.
```suggestion
BOOTMAGIC_ENABLE = lite  # Virtual DIP switch configuration(+1000)
```

File: keyboards/8pack/readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# 8-Pack Macropad
+
+An open source 2x4 macropad designed by Charles Garcia.
+
+Keyboard Maintainer: [Charles Garcia](https://github.com/cgarcia2097)
+Hardware Supported: 8-Pack Macropad PCB
```
Comment (e11i0t23) [inline keyboards/8pack/readme.md]:
```suggestion
Hardware Supported: 8-Pack Macropad PCB  
```

File: keyboards/8pack/readme.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# 8-Pack Macropad
+
+An open source 2x4 macropad designed by Charles Garcia.
+
+Keyboard Maintainer: [Charles Garcia](https://github.com/cgarcia2097)
```
Comment (e11i0t23) [inline keyboards/8pack/readme.md]: We need to add double space to end of lines in order for github to render the file properly

```suggestion
Keyboard Maintainer: [Charles Garcia](https://github.com/cgarcia2097)  
```

Comment (yanfali) [reply keyboards/8pack/readme.md]: I'm not a huge fan of double space, as editors have a habit of lopping it off. Much prefer explict <br>

File: keyboards/8pack/rules.mk (hunk 1 above)
Comment (gorbachev) [reply keyboards/8pack/rules.mk]: The macropad has a reset switch.

File: keyboards/8pack/readme.md (hunk 3 above)
Comment (gorbachev) [reply keyboards/8pack/readme.md]: look at that, it does indeed not render properly. I'll add the double spaces once I get back home after work.

File: keyboards/8pack/config.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#define PRODUCT_ID      0x2171
+#define DEVICE_VER      0x0002
+#define MANUFACTURER    Charles Garcia
+#define PRODUCT         8-Pack
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (e11i0t23) [inline keyboards/8pack/config.h]: This definition has had its name changed
```suggestion
#define DEBOUNCE 5
```

File: keyboards/8pack/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+ * rev 1.2 keymap
+ *
+ */
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	[0] = LAYOUT_REV12(
+	    KC_V, KC_C, KC_X, MO(1),
+		KC_A, KC_S, KC_D, KC_F
+	),
+	[1] = LAYOUT_REV12(
```
Comment (drashna) [inline keyboards/8pack/keymaps/default/keymap.c]:
```suggestion
	[1] = LAYOUT_rev12(
```

File: keyboards/8pack/keymaps/rev11/keymap.c [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#include QMK_KEYBOARD_H
+
+/*
+ * rev 1.1 keymap
+ *
+ */
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	[0] = LAYOUT_REV11(
```
Comment (drashna) [inline keyboards/8pack/keymaps/rev11/keymap.c]:
```suggestion
	[0] = LAYOUT_rev11(
```

File: keyboards/8pack/keymaps/rev11/keymap.c [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+	    KC_V, KC_C, KC_X, MO(1),
+		KC_A, KC_S, KC_D, KC_F
+	),
+	[1] = LAYOUT_REV11(
```
Comment (drashna) [inline keyboards/8pack/keymaps/rev11/keymap.c]:
```suggestion
	[1] = LAYOUT_rev11(
```

File: keyboards/8pack/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,17 @@
+#include QMK_KEYBOARD_H
+
+/*
+ * rev 1.2 keymap
+ *
+ */
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	[0] = LAYOUT_REV12(
```
Comment (drashna) [inline keyboards/8pack/keymaps/default/keymap.c]:
```suggestion
	[0] = LAYOUT_rev12(
```

File: keyboards/8pack/8pack.h [hunk 9]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_REV11( \
```
Comment (drashna) [inline keyboards/8pack/8pack.h]:
```suggestion
#define LAYOUT_rev11( \
```

File: keyboards/8pack/8pack.h [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    K00, K01, K02, K03, \
+	K10, K11, K12, K13 \
+) { \
+	{ K13, K12, K11, K10 }, \
+    { K03, K02, K01, K00 } \
+}
+
+#define LAYOUT_REV12( \
```
Comment (drashna) [inline keyboards/8pack/8pack.h]:
```suggestion
#define LAYOUT_rev12( \
```

**Comment by gorbachev** [comment]
> Actually, it may be better to create revision folders, for the keyboard.
> 
> That way, the same layout macro can be used, but the matrix is changed based on that.

This change is done and tested.

**Comment by e11i0t23** [comment]
You do not have a config.h in rev1.1 therefore USB versionID isnt defined when using rev1.1

File: keyboards/8pack/rules.mk [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (fauxpark) [inline keyboards/8pack/rules.mk]: Please replace this block with:

```make
# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   atmega32a    bootloadHID
BOOTLOADER = caterina
```

Comment (gorbachev) [reply keyboards/8pack/rules.mk]: Done

File: keyboards/8pack/rev11/config.h [hunk 12]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+
+#include QMK_KEYBOARD_CONFIG_H
```
Comment (drashna) [inline keyboards/8pack/rev11/config.h]: This isn't actually doesn't do anything.  It should be removed

```suggestion
```

File: keyboards/8pack/rev12/config.h [hunk 13]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+
+#include QMK_KEYBOARD_CONFIG_H
```
Comment (drashna) [inline keyboards/8pack/rev12/config.h]: This isn't actually doesn't do anything.  It should be removed

```suggestion
```

Comment (gorbachev) [reply keyboards/8pack/rev12/config.h]: Done

**Pull request merged**